spark/v4.0/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestWriteAborts.java

```diff
@@ -121,7 +121,7 @@ public void testBatchAppend() throws IOException {
         "Should be no orphan data files",
         ImmutableList.of(),
         sql(
-            "CALL %s.system.remove_orphan_files(table => '%s', older_than => %dL, location => '%s')",
+            "CALL %s.system.remove_orphan_files(table => '%s', older_than => CAST(%dL AS TIMESTAMP), location => '%s')",
```

Comment:
implicit cast does not work here, LONG => TIMESTAMP requries an explicitly cast

Reply:
this follows https://spark.apache.org/docs/latest/sql-ref-ansi-compliance.html

Reply:
oh nice, so disable ansi should make this case work?

Reply:
unfortunately, no, it does not satisfy the implicit cast requirements

Reply:
i see, too bad, that is another backward incompatbiility.

Reply:
Yeah it's unfortunate but it is a major version upgrade so I think it's OK. I also don't think the iceberg should get too involved in trying to work around spark behavior especially for a major version upgrade. 

I do think we should call these out in our procedure docs so that way we can make it easy for folks to figure out any gotcha's on upgrading